pkg/network/tracer/ebpf_conntracker.go

```diff
@@ -0,0 +1,330 @@
+// +build linux_bpf
+
+package tracer
+
+import "C"
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/network"
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/ebpf/probes"
+	"github.com/DataDog/datadog-agent/pkg/network/netlink"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/ebpf"
+	"github.com/DataDog/ebpf/manager"
+	ct "github.com/florianl/go-conntrack"
+	"golang.org/x/sys/unix"
+)
+
+/*
+#include "../ebpf/c/runtime/conntrack-types.h"
+*/
+import "C"
+
+const (
+	initializationTimeout = time.Second * 10
+)
+
+var tuplePool = sync.Pool{
+	New: func() interface{} {
+		return new(ConnTuple)
+	},
+}
+
+type conntrackTelemetry C.conntrack_telemetry_t
+
+type ebpfConntracker struct {
+	m            *manager.Manager
+	ctMap        *ebpf.Map
+	telemetryMap *ebpf.Map
+	// only kept around for stats purposes from initial dump
+	consumer *netlink.Consumer
+
+	stats struct {
+		gets                 int64
+		getTotalTime         int64
+		unregisters          int64
+		unregistersTotalTime int64
+	}
+}
+
+// NewEBPFConntracker creates a netlink.Conntracker that monitor conntrack NAT entries via eBPF
+func NewEBPFConntracker(cfg *config.Config) (netlink.Conntracker, error) {
+	buf, err := getRuntimeCompiledConntracker(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to compile ebpf conntracker: %s", err)
+	}
+
+	m, err := getManager(buf, cfg.ConntrackMaxStateSize)
+	if err != nil {
+		return nil, err
+	}
+
+	err = m.Start()
+	if err != nil {
+		_ = m.Stop(manager.CleanAll)
+		return nil, fmt.Errorf("failed to start ebpf conntracker: %s", err)
+	}
+
+	ctMap, _, err := m.GetMap(string(probes.ConntrackMap))
+	if err != nil {
+		_ = m.Stop(manager.CleanAll)
+		return nil, fmt.Errorf("unable to get conntrack map: %s", err)
+	}
+
+	telemetryMap, _, err := m.GetMap(string(probes.ConntrackTelemetryMap))
+	if err != nil {
+		_ = m.Stop(manager.CleanAll)
+		return nil, fmt.Errorf("unable to get telemetry map: %s", err)
+	}
+
+	e := &ebpfConntracker{
+		m:            m,
+		ctMap:        ctMap,
+		telemetryMap: telemetryMap,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), initializationTimeout)
+	defer cancel()
+
+	err = e.dumpInitialTables(ctx, cfg)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			return nil, fmt.Errorf("could not initialize conntrack after %s", initializationTimeout)
```

Comment:
```suggestion
			return nil, fmt.Errorf("could not initialize conntrack after %w", initializationTimeout)
```

Reply:
This one isn't an error, but a `time.Duration`.